samcli/commands/init/init_templates.py

```diff
@@ -171,3 +180,75 @@ def is_dynamic_schemas_template(self, package_type, app_template, runtime, base_
             if option.get("appTemplate") == app_template:
                 return option.get("isDynamicTemplate", False)
         return False
+
+    def get_preprocessed_manifest(self, filter_value=None):
+        """
+        This method get the manifest cloned from the git repo and preprocessed it.
+        Below is the link to manifest:
+        https://github.com/aws/aws-sam-cli-app-templates/blob/master/manifest.json
+
+        The structure of the manifest is shown below:
+        {
+            "dotnetcore2.1": [
+                {
+                    "directory": "dotnetcore2.1/cookiecutter-aws-sam-hello-dotnet",
+                    "displayName": "Hello World Example",
+                    "dependencyManager": "cli-package",
+                    "appTemplate": "hello-world",
+                    "packageType": "Zip",
+                    "useCaseName": "Serverless API"
+                },
+            ]
+        }
+
+        Parameters
+        ----------
+        filter_value : string, optional
+            This could be a runtime or a base-image, by default None
+
+        Returns
+        -------
+        [dict]
+            This is preprocessed manifest with the use_case as key
+        """
+        self.clone_templates_repo()
+        manifest_path = Path(self._git_repo.local_path, "manifest.json")
```

Comment:
we should have any string constants such as "manifest.json" as class level constants.